### PR TITLE
Sequence: Support for JSON reports

### PR DESCRIPTION
**Pull request by psiinon**
## Overview
As per title.
Note that I didnt add the 'highestAlert' field - that looks to be a bit tricky, and not really needed for JSON reports?
Might want it for HTML ones though 😁 
Also added the alertId so that they can be cross referenced.

## Checklist
- [ ] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [ ] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

**Comment by thc202** [comment]
This has conflicts now.

**Comment by psiinon** [comment]
Ready to review, but I have found one failing edge case that I'm looking into..

**Comment by psiinon** [comment]
OK, the edge case was Zest not handling a req with a token in the URL - now fixed (with a test too!)

File: addOns/reports/src/test/resources/org/zaproxy/addon/reports/resources/basic-traditional-json-plus.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
                     "desc": "<p>XSS Description</p>",
                     "instances":[ 
                         {
+							"id": "-1",
```
Comment (kingthorin) [inline addOns/reports/src/test/resources/org/zaproxy/addon/reports/resources/basic-traditional-json-plus.json:23]: Should we care about the formatting here?
L36 too

File: addOns/sequence/src/main/resources/org/zaproxy/zap/extension/sequence/resources/Messages.properties [hunk 2]
```diff
@@ -20,6 +20,7 @@ sequence.automation.import.sequencecreated = Job {0}: Created sequence with {1}
 sequence.automation.import.summary = Name: {0}, Path: {1}
 sequence.automation.name = Sequence Automation
 sequence.automation.step = Step {0}
+sequence.automation.step.pass = Pass
```
Comment (kingthorin) [inline addOns/sequence/src/main/resources/org/zaproxy/zap/extension/sequence/resources/Messages.properties:23]: We don't need a fail label? That seems strange.

File: addOns/zest/src/main/java/org/zaproxy/zap/extension/zest/ZestZapUtils.java [hunk 3]
```diff
@@ -1031,6 +1035,7 @@ public static HttpMessage toHttpMessage(ZestRequest request, ZestResponse respon
                 msg.setResponseHeader(new HttpResponseHeader(response.getHeaders()));
             } catch (Exception e) {
                 LOGGER.error(e.getMessage(), e);
+                e.printStackTrace(); // TODO
```
Comment (kingthorin) [inline addOns/zest/src/main/java/org/zaproxy/zap/extension/zest/ZestZapUtils.java:1038]: :wink:

Comment (psiinon) [reply addOns/zest/src/main/java/org/zaproxy/zap/extension/zest/ZestZapUtils.java:1038]: Arg!

File: addOns/sequence/src/main/resources/org/zaproxy/zap/extension/sequence/resources/Messages.properties (hunk 2 above)
Comment (psiinon) [reply addOns/sequence/src/main/resources/org/zaproxy/zap/extension/sequence/resources/Messages.properties:23]: Because the failure message comes from the assertion, and thats already i18ned

**Comment by psiinon** [comment]
Now with unit tests 😁

**Review by thc202 — APPROVED** [review]

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
Do we want to update the help for these?
- https://www.zaproxy.org/docs/desktop/addons/report-generation/report-traditional-json-plus/
- https://www.zaproxy.org/docs/desktop/addons/report-generation/report-traditional-json/
and for that matter the HTML variants when I get them finished.

**Comment by psiinon** [comment]
@kingthorin yes, definitely. Can you do that? 😁

**Comment by kingthorin** [comment]
Yup, no problem.
